libretroshare/src/util/i2pcommon.cpp

```diff
@@ -0,0 +1,174 @@
+#include "i2pcommon.h"
+
+#include "util/rsbase64.h"
+#include "util/rsdebug.h"
+
+namespace i2p {
+
+const std::string generateNameSuffix(const size_t len) {
+	std::vector<uint8_t> tmp(len);
+	RsRandom::random_bytes(tmp.data(), len);
+	const std::string location = Radix32::encode(tmp.data(), len);
+
+	return location;
+}
+
+std::string keyToBase32Addr(const std::string &key)
+{
+	std::string copy(key);
+
+	// replace I2P specific chars
+	std::replace(copy.begin(), copy.end(), '~', '/');
+	// replacing the - with a + is not necessary, as RsBase64 can handle base64url encoding, too
+	// std::replace(copy.begin(), copy.end(), '-', '+');
+
+	// decode
+	std::vector<uint8_t> bin;
+	RsBase64::decode(copy, bin);
+
+	// hash
+	std::vector<uint8_t> sha256 = RsUtil::BinToSha256(bin);
+	// encode
+	std::string out = Radix32::encode(sha256);
+
+	// i2p uses lowercase
+	std::transform(out.begin(), out.end(), out.begin(), ::tolower);
+	out.append(".b32.i2p");
+
+	return out;
+}
+
+const std::string makeOption(const std::string &lhs, const int8_t &rhs) {
+	// convert number to int
+	std::ostringstream oss;
+	oss << (int)rhs;
```

Comment:
You can also use `std::to_string` instead so it becomes just one line ;)

Reply:
done :+1: